Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Utils/IndexableList.h

```diff
@@ -120,6 +120,7 @@ namespace AZ
 
                 m_data[positionToRemove].m_next = m_freeListHead;
                 m_freeListHead = positionToRemove;
+                m_data[positionToRemove].m_value.~T();
```

Comment:
Is that really safe to do?
This will call the destructor again once the `IndexableList` is destroyed and the vector is cleaned up, which should not happen.

Reply:
Ah sorry, overlooked the second commit 👍